backpack/core/derivatives/avgpoolnd.py

```diff
@@ -31,11 +33,29 @@ def __init__(self, N):
             self.conv = Conv3d
             self.convt = ConvTranspose3d
 
+    def check_parameters(self, module: Module) -> None:
+        assert module.count_include_pad, (
+            "Might not work for exotic hyperparameters of AvgPool2d, "
+            + "like count_include_pad=False"
+        )
+
+    def get_avg_pool_parameters(self, module) -> Tuple[Any, Any, Any]:
+        """Return the parameters of the module.
+
+        Args:
+            module: module
+
+        Returns:
+            stride, kernel_size, padding
+        """
+        return module.stride, module.kernel_size, module.padding
+
     def hessian_is_zero(self):
         return True
 
     def ea_jac_t_mat_jac_prod(self, module, g_inp, g_out, mat):
         """Use fact that average pooling can be implemented as conv."""
+        self.check_parameters(module)
         if self.N == 1:
```

Comment:
Refactoring suggestion to avoid the `raise NotImplementedError` (I think it's there because otherwise variables may be undefined below): Compute `C, in_features, out_features, shape_out` independent of `self.N`.
```python
C = module.input0.shape[1]
shape_out = (1,) + tuple(module.input0.shape[2:])
in_features = module.input0.shape[2:].numel()
out_features = module.output.shape[2:].numel()
```

Reply:
Refactored with slightly different version
```
C = module.input0.shape[1]
shape_out = (1,) + tuple(module.output.shape[2:])
in_features = module.input0.shape[1:].numel()
out_features = module.output.shape[1:].numel()
```